interfaces/builtin/network_manager.go

```diff
@@ -0,0 +1,444 @@
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.Device.WiMax"/>
+
+    <!-- Devices (read/write, secured with PolicyKit) -->
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.Device.Wireless"/>
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.Device"/>
+
+    <!-- Core stuff (read-only properties, no methods) -->
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.Connection.Active"/>
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.DHCP4Config"/>
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.DHCP6Config"/>
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.IP4Config"/>
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.IP6Config"/>
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.VPN.Connection"/>
+
+    <!-- Core stuff (read/write, secured with PolicyKit) -->
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager"/>
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.Settings"/>
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.Settings.Connection"/>
+
+    <!-- Agents; secured with PolicyKit.  Any process can talk to
+         the AgentManager API, but only NetworkManager can talk
+         to the agents themselves. -->
+    <allow send_destination="org.freedesktop.NetworkManager"
+           send_interface="org.freedesktop.NetworkManager.AgentManager"/>
+
+    <!-- Root-only functions -->
+    <deny send_destination="org.freedesktop.NetworkManager"
+          send_interface="org.freedesktop.NetworkManager"
+          send_member="SetLogging"/>
+    <deny send_destination="org.freedesktop.NetworkManager"
+          send_interface="org.freedesktop.NetworkManager"
+          send_member="Sleep"/>
+    <deny send_destination="org.freedesktop.NetworkManager"
+          send_interface="org.freedesktop.NetworkManager.Settings"
+          send_member="LoadConnections"/>
+    <deny send_destination="org.freedesktop.NetworkManager"
+          send_interface="org.freedesktop.NetworkManager.Settings"
+          send_member="ReloadConnections"/>
+
+    <deny own="org.freedesktop.NetworkManager.dnsmasq"/>
+    <deny send_destination="org.freedesktop.NetworkManager.dnsmasq"/>
+</policy>
```

Comment:
Noting for thoroughness, 16.04 network manager has `<policy user="whoopsie">`. Since whoopsie is not packaged as a snap and we don't have per-snap uids, this is not an issue. Otherwise, the bus policy looks fine.